src/probnum/diffeq/odefiltsmooth/ivpfiltsmooth.py

```diff
@@ -17,28 +24,172 @@ class GaussianIVPFilter(ODESolver):
 
     Parameters
     ----------
-    gaussfilt : gaussianfilter.GaussianFilter
-        e.g. the return value of ivp_to_ukf(), ivp_to_ekf1().
-
-    Notes
-    -----
-    - gaussfilt.dynamicmodel contains the prior,
-    - gaussfilt.measurementmodel contains the information about the ODE right hand side function,
-    - gaussfilt.initialdistribution contains the information about the initial values.
+    ivp
+        Initial value problem to be solved.
+    prior
+        Prior distribution.
+    measurement_model
+        ODE measurement model.
+    with_smoothing
+        To smooth after the solve or not to smooth after the solve.
+    init_implementation
+        Initialization algorithm. Either via Scipy (``initialize_odefilter_with_rk``) or via Taylor-mode AD (``initialize_odefilter_with_taylormode``).
+        For more convenient construction, consider :func:`GaussianIVPFilter.from_rk_init`.
     """
 
-    def __init__(self, ivp, gaussfilt, with_smoothing):
-        if not isinstance(gaussfilt.dynamics_model, pnfs.statespace.Integrator):
+    def __init__(
+        self,
+        ivp: IVP,
+        prior: pnfs.statespace.Integrator,
+        measurement_model: pnfs.statespace.DiscreteGaussian,
+        with_smoothing: bool,
+        init_implementation: typing.Callable[
+            [typing.Callable, np.ndarray, float, pnfs.statespace.Integrator], Normal
+        ],
+    ):
+
+        initrv = Normal(
+            np.zeros(prior.dimension),
+            np.zeros((prior.dimension, prior.dimension)),
+            cov_cholesky=np.zeros((prior.dimension, prior.dimension)),
+        )
+        self.gfilt = pnfs.Kalman(
+            dynamics_model=prior, measurement_model=measurement_model, initrv=initrv
+        )
+
+        if not isinstance(prior, pnfs.statespace.Integrator):
             raise ValueError(
                 "Please initialise a Gaussian filter with an Integrator (see filtsmooth.statespace)"
             )
-        self.gfilt = gaussfilt
         self.sigma_squared_mle = 1.0
         self.with_smoothing = with_smoothing
-        super().__init__(ivp=ivp, order=gaussfilt.dynamics_model.ordint)
+        self.init_implementation = init_implementation
+        super().__init__(ivp=ivp, order=prior.ordint)
+
+    @staticmethod
+    def string_to_measurement_model(
+        measurement_model_string, ivp, prior, measurement_noise_covariance=0.0
+    ):
+        """Construct a measurement model :math:`\\mathcal{N}(g(m), R)` for an ODE.
```

Comment:
this replaces everything that used to be in `ivp2filter`. Once the PR for the probsolve_ivp interface is done, this function will be much leaner.

I decided to move it here as a staticmethod to make absolutely clear that this method belongs to the ODE filter.

Reply:
Looks fine :+1: In case this change was not necessary for this initialization PR, I think it would have been better to include it in the other PR, just to keep this one strictly on-topic. But it's still a good change of course :)

Reply:
Since we decided to move the initialisation into `ODESolver.initialize()`, the __init__ had to be thought through again in terms of taking prior and measurement model (not initrv). since the `ivp2filter` functions returned the full filter, I considered it easier to write this function instead of refactoring them in order to return the correct things (which would've been removed in the next step anyway).